### PR TITLE
Add mobile menu subcategories

### PR DESCRIPTION
**Pull request by jose-correia**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Screenshot 2023-04-28 at 17 22 22

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/
- After: https://mobile-menu-subcategories--moleculardevices--hlxsites.hlx.page/

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -591,6 +630,53 @@ header .mobile-menu .request-quote a {
...
+
+header .mobile-menu .back-to-parent:hover {
+  background: var(--background-color);
+}
+
+header .mobile-menu .mobile-menu-item.submenu-open .mobile-menu-subcategories {
+  display: block;
+}
+
+header .mobile-menu .caret {
```
Comment (paolomoz) [inline blocks/header/header.css:648]: the chevron is not aligned with the circle
Screenshot 2023-05-02 at 10 29 12

File: blocks/header/menus/mobile-menu.js [hunk 2]
```diff
@@ -9,6 +10,38 @@ import {
...
+      hamburger.classList.remove('hamburger-open');
+      hamburger.classList.add('hamburger-close');
+      mobileMenu.classList.toggle('mobile-menu-open');
+      body.classList.toggle('openmenu');
+    } else {
+      // if hamburger is closed, open it
+      hamburger.classList.remove('hamburger-close');
+      hamburger.classList.add('hamburger-open');
+      mobileMenu.classList.toggle('mobile-menu-open');
+      body.classList.toggle('openmenu');
```
Comment (paolomoz) [inline blocks/header/menus/mobile-menu.js:30]: are all these class toggles needed? couldn't it be managed with only 1 class assigned to an ancestor, that all descendants can use in css selectors?

File: blocks/header/header.css [hunk 3]
```diff
@@ -278,6 +279,12 @@ header .mobile-menu li:hover {
   background: #f0f1f2;
 }
 
+header .mobile-menu .mobile-menu-items li {
+  line-height: 1.4286;
+  font-size: 18px;
+  border-bottom: 1px solid #f1f2f3;
```
Comment (paolomoz) [inline blocks/header/header.css:285]: .headersearch-item should not have border bottom
Screenshot 2023-05-02 at 10 35 11

**Review by paolomoz — COMMENTED** [review]
in original the logo and the hamburger icon have space left and right (not at the full left and full right of the window)

Screenshot 2023-05-02 at 10 38 32

File: blocks/header/menus/mobile-menu.js (hunk 2 above)
Comment (jose-correia) [reply blocks/header/menus/mobile-menu.js]: Very good point! I could use the class in the body to style everything

**Review by lydiapuric — APPROVED** [review]
+1 LGTM

**Pull request merged**